extensions/ql-vscode/src/query-serialization.ts

```diff
@@ -12,161 +12,167 @@ import { QueryHistoryInfo } from "./query-history/query-history-info";
 import { QueryStatus } from "./query-status";
 import { QueryEvaluationInfo } from "./run-queries-shared";
 import { QueryResultType } from "./pure/legacy-messages";
-import { JsonValidator } from "./data-serialization/json-validator";
 import { RemoteQueryHistoryItem } from "./data-serialization/source-schemas-types/remote-query-history-item";
 import { VariantAnalysisHistoryItem } from "./data-serialization/source-schemas-types/variant-analysis-history-item";
+import { App } from "./common/app";
+import { DisposableObject } from "./pure/disposable-object";
 
 type SerializableHistoryItems =
   | LocalQueryInfo
   | RemoteQueryHistoryItem
   | VariantAnalysisHistoryItem;
 
-export async function deserializeQueryHistory(
-  fsPath: string,
-): Promise<SerializableHistoryItems[]> {
-  try {
-    if (!(await pathExists(fsPath))) {
-      return [];
-    }
+export class QueryHistorySerializer extends DisposableObject {
```

Comment:
I think we should rename the file to `query-history-serializer.ts` since that is what it now contains.

Reply:
Thanks for keeping me honest 💐